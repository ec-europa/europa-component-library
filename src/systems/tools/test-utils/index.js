const merge = require('deepmerge');
const twing = require('@ecl-twig/ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const test = document.createElement('test');
      test.innerHTML = html.trim();
      resolve(test);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
