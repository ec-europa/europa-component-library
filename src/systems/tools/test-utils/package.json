{
  "name": "@ecl-twig/test-utils",
  "version": "2.31.1",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
