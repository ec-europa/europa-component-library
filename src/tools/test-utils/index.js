const merge = require('deepmerge');
const twing = require('@ecl/twig-ec-storybook/.storybook/environment');

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) => {
    try {
      const html = twing.render(file, options);
      const div = document.createElement('div');
      div.innerHTML = html.trim();
      resolve(div.firstChild);
    } catch (error) {
      reject(error);
    }
  });

module.exports = {
  merge,
  renderTwigFileAsNode,
};
