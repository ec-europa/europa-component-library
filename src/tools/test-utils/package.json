{
  "name": "@ecl/test-utils",
  "version": "3.4.2",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
