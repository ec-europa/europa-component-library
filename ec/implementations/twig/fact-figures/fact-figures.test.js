import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

// Import data for tests
import data3Col from './demo/data--3-col';
import data4Col from './demo/data--4-col';

describe('EC - Fact and figures', () => {
  const template =
    '@ecl-twig/ec-component-fact-figures/ecl-fact-figures.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('3 cols', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(data3Col)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data3Col, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data3Col, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('4 cols', () => {
    test('renders correctly', () => {
      expect.assertions(1);

      return expect(render(data4Col)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data4Col, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data4Col, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
  describe('with validation enabled and missing input data', () => {
    test('returns the right warning message', () => {
      expect.assertions(1);
      const dataCompliance = { ...data3Col, _compliance_: true };
      dataCompliance.items[0].value = '';

      return expect(render(dataCompliance)).resolves.toMatchSnapshot();
    });
  });
});
