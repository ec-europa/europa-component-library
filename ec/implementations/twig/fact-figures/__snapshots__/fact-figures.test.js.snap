// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Fact and figures 3 cols renders correctly 1`] = `
<test>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-3"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aliquam faucibus nulla eget eleifend
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Nam lacinia nisl eget diam mattis
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Sed efficitur bibendum rutrum. Nunc feugiat congue augue ac consectetur.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-fact-figures__view-all-link"
        href="/example"
      >
        View all metrics
      </a>
    </div>
  </div>
</test>
`;

exports[`EC - Fact and figures 3 cols renders correctly with extra attributes 1`] = `
<test>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-3"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aliquam faucibus nulla eget eleifend
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Nam lacinia nisl eget diam mattis
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Sed efficitur bibendum rutrum. Nunc feugiat congue augue ac consectetur.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-fact-figures__view-all-link"
        href="/example"
      >
        View all metrics
      </a>
    </div>
  </div>
</test>
`;

exports[`EC - Fact and figures 3 cols renders correctly with extra class names 1`] = `
<test>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-3 custom-class custom-class--test"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aliquam faucibus nulla eget eleifend
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Nam lacinia nisl eget diam mattis
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Sed efficitur bibendum rutrum. Nunc feugiat congue augue ac consectetur.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-fact-figures__view-all-link"
        href="/example"
      >
        View all metrics
      </a>
    </div>
  </div>
</test>
`;

exports[`EC - Fact and figures 4 cols renders correctly 1`] = `
<test>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-4"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-fact-figures__view-all-link"
        href="/example"
      >
        View all metrics
      </a>
    </div>
  </div>
</test>
`;

exports[`EC - Fact and figures 4 cols renders correctly with extra attributes 1`] = `
<test>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-4"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-fact-figures__view-all-link"
        href="/example"
      >
        View all metrics
      </a>
    </div>
  </div>
</test>
`;

exports[`EC - Fact and figures 4 cols renders correctly with extra class names 1`] = `
<test>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-4 custom-class custom-class--test"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-fact-figures__view-all-link"
        href="/example"
      >
        View all metrics
      </a>
    </div>
  </div>
</test>
`;

exports[`EC - Fact and figures with validation enabled and missing input data returns the right warning message 1`] = `
<test>
  <div
    class="ecl-fact-figures ecl-fact-figures--col-3"
  >
    <div
      class="ecl-fact-figures__items"
    >
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[icon] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
        <div
          class="ecl-fact-figures__title"
        >
          Lorem ipsum
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Nunc condimentum sapien ut nibh finibus suscipit vitae at justo. Morbi quis odio faucibus, commodo tortor id, elementum libero.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[icon] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Sed hendrerit
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Turpis varius congue venenatis, erat dui feugiat felis.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[icon] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Donec suscipit interdum augue, ac dapibus eros finibus a.
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Cras vestibulum efficitur mi, quis porta tellus rutrum ut. Quisque at pulvinar sem.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[icon] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aenean dapibus
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Aliquam lacinia diam eu sem malesuada, in interdum ante bibendum.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[icon] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Aliquam faucibus nulla eget eleifend
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Duis nec lectus tortor.
        </div>
      </div>
      <div
        class="ecl-fact-figures__item"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--m ecl-fact-figures__icon"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#general--digital"
          />
        </svg>
        <span
          class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
          title="[icon] - ecl compliant"
        >
          <svg
            aria-hidden="true"
            class="ecl-icon ecl-icon--xs"
            focusable="false"
          >
            <use
              xlink:href="/icons.svg#notifications--success"
            />
          </svg>
        </span>
        <div
          class="ecl-fact-figures__value"
        >
          00.0 million
        </div>
        <div
          class="ecl-fact-figures__title"
        >
          Nam lacinia nisl eget diam mattis
        </div>
        <div
          class="ecl-fact-figures__description"
        >
          Sed efficitur bibendum rutrum. Nunc feugiat congue augue ac consectetur.
        </div>
      </div>
    </div>
    <div
      class="ecl-fact-figures__view-all"
    >
      <a
        class="ecl-link ecl-link--standalone ecl-fact-figures__view-all-link"
        href="/example"
      >
        View all metrics
      </a>
      <span
        class="ecl-u-type-color-green ecl-compliance-inner ecl-compliance-success ecl-u-ma-2xs"
        title="[link--standalone] - ecl compliant"
      >
        <svg
          aria-hidden="true"
          class="ecl-icon ecl-icon--xs"
          focusable="false"
        >
          <use
            xlink:href="/icons.svg#notifications--success"
          />
        </svg>
      </span>
    </div>
  </div>
  <!-- ecl2-compliance: fact-figures -->
  <div
    class="ecl2-compliance ecl-u-border-width-1 ecl-u-border-color-red ecl-u-border-style-solid ecl-u-ma-xs ecl-u-pr-s ecl-u-bg-white"
  >
    <p
      class="ecl-u-type-color-grey ecl-u-ml-m ecl-u-type-xs"
    >
      [fact-figures] -  The {{ value }} in each item needs to be set and filled in.
    </p>
  </div>
</test>
`;
